### PR TITLE
Korefe 210 handle casing internally

### PR DESCRIPTION
**Pull request by robwierzbowski**
## Problem

LogWeasel expects a CONSTANT_CASE version of the app name to be passed into the client side package. Almost every other client side tool we use expects a dashed-case name, just like the canonical fixops app name. Users are making mistakes, passing in a dashed case name instead of a constant case name, or spending time researching, reviewing, and fixing PRs to avoid the mistake.

In addition, there's no connection between the string that the user passes into the client side tool, and the `app_name` that the gem consumes by default. It's left to the user to make those match, or to even realize they should.

## Solution

Allow the user to pass in the canonical fixops dash-case app name, just like the other client side tools that ask for app names.

Document pulling the canonical fixops app name from package.json. In almost all cases, this should match the `app_name` that the Rails code is consuming (but in dashed case).

Convert the passed in app name to constant case to exactly match the `app_name` that Rails is providing by default.

## And a little extra

I also improved the readme a little, switching to markdown and adding syntax highlighting. I can pull that into a separate PR if desired.

### Before Merging

- [ ] If there is an RC on this branch, revert the version change in `version.rb`

### After Merging

See the [gem release process](https://github.com/stitchfix/eng-wiki/blob/master/technical-topics/updating-gem-versions.md) for a detailed list, but the gist of it is:

- [ ] Fetch `master` locally and run the applicable `rake version:*` task **on `master`** to bump the version
- [ ] Run `rake release` **on `master`** to release the new version on Gemfury
- [ ] Add [release notes](https://github.com/stitchfix/log_weasel/releases) - **this is very important in helping other engineers understand what changed in the new version**

### CODE REVIEW

File: src/stitch_fix/__tests__/logWeaselTest.test.js [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
-    expect(() => LogWeasel.init()).toThrow();
+  it('throws an error if key is not passed', () => {
+    expect(() => LogWeasel.init()).toThrowErrorMatchingSnapshot();
   });
 });
 
 describe('generateId', () => {
   beforeAll(() => {
-    LogWeasel.init('WEASEL');
+    LogWeasel.init('weasel-gang-service');
```
Comment (robwierzbowski) [inline src/stitch_fix/__tests__/logWeaselTest.test.js:11]: https://www.google.com/search?q=what+is+a+group+of+weasels+called

**Comment by robwierzbowski** [comment]
Not sure what's going on with the failed Ruby tests, but they are not related to the code change in this PR.

File: README.md [hunk 2]
```diff
@@ -8,21 +8,21 @@ applications and application instances.
 
 Add log_weasel to your Gemfile:
 
-<pre>
```
Comment (brettfishman) [inline README.md:11]: Thank you for cleaning up the formatting in this file. Love having FE expertise on the team! 😂

File: src/stitch_fix/__tests__/logWeaselTest.test.js (hunk 1 above)
Comment (brettfishman) [reply src/stitch_fix/__tests__/logWeaselTest.test.js:11]: Personally I like "boogle" but "gang" works.

File: src/stitch_fix/logWeasel.js [hunk 3]
```diff
@@ -1,21 +1,18 @@
...
-    globalKey = key;
+    globalKey = constantCase(key);
   },
 
-  generateId: () => {
-    const formattedKey = `${globalKey}-JS`;
-
-    return `${ulid()}-${formattedKey}`;
-  },
+  generateId: () => `${ulid()}-${globalKey}-JS`,
```
Comment (brettfishman) [inline src/stitch_fix/logWeasel.js:15]: So much cleaner

File: README.md [hunk 4]
```diff
@@ -32,57 +32,60 @@ See [this setup guide](https://github.com/stitchfix/eng-wiki/blob/master/technic
...
 
 ## Javascript
 
 If you have added the optional NPM package, this is how you would use it:
 
-<pre>
+```js
 import LogWeasel from "@stitch-fix/log-weasel";
+// Get the canonical fixops app name from your app's package.json
+import { name } from '../../package.json';
```
Comment (brettfishman) [inline README.md:50]: ❤️  forgot about using the name defined in `package.json` - nice.

Comment (brettfishman) [reply README.md:50]: Wait, is there always an app name defined in `package.json`?

**Comment by brettfishman** [comment]
The test failures look related to Ruby 3.0. I will take a closer look soon.

File: README.md (hunk 4 above)
Comment (juliepagano) [reply README.md:50]: Looks like it's required. If someone is missing one, that's definitely something they should fix.

https://docs.npmjs.com/creating-a-package-json-file#required-name-and-version-fields

File: src/stitch_fix/__tests__/logWeaselTest.test.js (hunk 1 above)
Comment (juliepagano) [reply src/stitch_fix/__tests__/logWeaselTest.test.js:11]: I like a "confusion" of weasels. 🤣

**Review by juliepagano — COMMENTED** [review]
This is looking good on a quick first pass. I'll give it a closer look on monday when I am hopefully less full of allergy doom.

File: src/stitch_fix/__tests__/logWeaselTest.test.js (hunk 1 above)
Comment (robwierzbowski) [reply src/stitch_fix/__tests__/logWeaselTest.test.js:11]: They're all pretty good 😂

File: README.md (hunk 4 above)
Comment (robwierzbowski) [reply README.md:50]: @brettfishman And when using the app template, I believe the name in package.json will match the app's name in fixops!

**Pull request merged**